src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentProcessIdFixer.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>CA1837: Use Environment.ProcessId</summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseEnvironmentProcessIdFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UseEnvironmentProcessId.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            SemanticModel model = await doc.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode root = await doc.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span) is SyntaxNode node &&
```

Comment:
I believe this will return false when the diagnostic is on an argument: ‘SomeMethod(Process.GetCurrentProcess().Id);’. Can you please also add a unit test for this case?

```suggestion
            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node &&
```

Reply:
There is a unit test for it:
https://github.com/dotnet/roslyn-analyzers/pull/3838/files#diff-426f1f0417289c59e42bd1ebdc2718e6R85
It's why I added:
https://github.com/dotnet/roslyn-analyzers/pull/3838/files#diff-c0c60dc8311eeaa510154fcb87e26c2eR43
Maybe I can delete that check if I follow your getInnermostNodeForTie suggestion?

Reply:
Ah, I see you handle this case below and also have a unit test. You may be able to simplify the fixer code by still passing the above flag and not have to deal with IArgumentOperation below.

Reply:
If you make that change, ‘IOperation operation’ can be changed to ‘IPropertyReferenceOperation operation’

Reply:
Cool, that worked. Thanks.